diff --git a/celt/_kiss_fft_guts.h b/celt/_kiss_fft_guts.h
--- a/celt/_kiss_fft_guts.h
+++ b/celt/_kiss_fft_guts.h
@@ -119,6 +119,13 @@
           (m).i = SUB32_ovflw(S_MUL((a).i,(b).r) , S_MUL((a).r,(b).i)); }while(0)
 #endif
 
+#if defined(M61_OPUS_E907_FFT_COMPLEX_MAC)
+#undef C_MULC
+#define C_MULC(m,a,b) \
+   do { (m).r = ADD32_ovflw(S_MUL((a).r,(b).r), S_MUL((a).i,(b).i)); \
+        (m).i = SUB32_ovflw(S_MUL((a).i,(b).r), S_MUL((a).r,(b).i)); } while (0)
+#endif
+
 #   define C_MULBYSCALAR( c, s ) \
       do{ (c).r =  S_MUL( (c).r , s ) ;\
          (c).i =  S_MUL( (c).i , s ) ; }while(0)
