diff --git a/celt/ecintrin.h b/celt/ecintrin.h
--- a/celt/ecintrin.h
+++ b/celt/ecintrin.h
@@ -46,0 +47,11 @@
+
+#if defined(M61_OPUS_E907_CLZ32)
+static inline int m61_opus_e907_clz32(unsigned value)
+{
+  unsigned result;
+  __asm__ volatile("clz32 %0, %1" : "=r"(result) : "r"(value));
+  return (int)result;
+}
+# define EC_CLZ0    (32)
+# define EC_CLZ(_x) (m61_opus_e907_clz32(_x))
+#endif
@@ -51 +62 @@
-#if defined(_MSC_VER) && (_MSC_VER >= 1400)
+#if !defined(EC_CLZ) && defined(_MSC_VER) && (_MSC_VER >= 1400)
@@ -67 +78 @@
-#elif __GNUC_PREREQ(3,4)
+#elif !defined(EC_CLZ) && __GNUC_PREREQ(3,4)
