diff --git a/celt/bands.c b/celt/bands.c
--- a/celt/bands.c
+++ b/celt/bands.c
@@ -1394,6 +1394,9 @@ static void special_hybrid_folding(const CELTMode *m, celt_norm *norm, celt_no
       OPUS_COPY(&norm2[n1], &norm2[2*n1 - n2], n2-n1);
 }
 
+#if defined(M61_OPUS_TCM_QUANT_ALL_BANDS)
+__attribute__((section(".tcm_code.opus.quant_all_bands")))
+#endif
 void quant_all_bands(int encode, const CELTMode *m, int start, int end,
       celt_norm *X_, celt_norm *Y_, unsigned char *collapse_masks,
       const celt_ener *bandE, int *pulses, int shortBlocks, int spread,
